src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -2566,6 +2566,33 @@ static QVariant fcnGeomIsValid( const QVariantList &values, const QgsExpressionC
   return QVariant( isValid );
 }
 
+static QVariant fcnGeomMakeValid( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  if ( geom.isNull() )
+    return QVariant();
+
+  QgsGeometry valid = geom.makeValid();
+
+  return QVariant::fromValue( valid );
+}
+
+static QVariant fcnGeoCollAsArray( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  if ( geom.isNull() )
+    return QVariant();
+
+  QgsGeometryCollection multiGeom = geom.asGeometryCollection();
```

Comment:
did you compile this? :thinking:

Reply:
Yes and tested it, though I might have forgotten to take the latest files from my VM to upload, I'll validate again but the functions worked as intended locally.

Reply:
Looks better now, thanks!